src/components/ExportDataModal.js

```diff
@@ -0,0 +1,228 @@
+import { icon } from 'src/components/icons'
+import Modal from 'src/components/Modal'
+import { withWorkspaces, WorkspaceSelector } from 'src/components/workspace-utils'
+import { ajaxCaller } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
+import { reportError } from 'src/libs/error'
+import { requiredFormLabel, formLabel } from 'src/libs/forms'
+import * as Nav from 'src/libs/nav'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+import { Component } from 'src/libs/wrapped-components'
+import validate from 'validate.js'
+
+
+class infoTile extends Component {
+  static propTypes = {
+    infoStyle: PropTypes.object.isRequired,
+    content: PropTypes.string.isRequired,
+    iconName: PropTypes.string.isRequired
+  }
+
+  render() {
+    const { infoStyle, content, iconName } = this.props
+    return div({ style: { ...infoStyle, display: 'flex', alignItems: 'center' } }, [
+      icon(iconName, { size: 36, className: 'is-solid', style: { flex: 'none', marginRight: '0.5rem' } }),
+      content
+    ])
+  }
+}
+
+
+export default _.flow(
+  ajaxCaller,
+  withWorkspaces()
+)(class ExportDataModal extends Component {
+  static propTypes = {
+    onDismiss: PropTypes.func.isRequired,
+    selectedEntities: PropTypes.array.isRequired,
+    selectedDataType: PropTypes.string.isRequired,
+    runningSubmissionsCount: PropTypes.number.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      hardConflicts: [],
+      softConflicts: [],
+      additionalDeletions: [],
+      selectedWorkspaceId: undefined,
+      error: undefined,
+      copying: false,
+      selectedEntityType: props.selectedDataType
+    }
+  }
+
+  getSelectedWorkspace() {
+    const { workspaces } = this.props
+    const { selectedWorkspaceId } = this.state
+    return _.find({ workspace: { workspaceId: selectedWorkspaceId } }, workspaces)
+  }
+
+  render() {
+    const { copied } = this.state
+    return copied ? this.renderPostCopy() : this.renderCopyForm()
+  }
+
+  renderCopyForm() {
+    const { onDismiss, selectedEntities, runningSubmissionsCount, workspace, workspaces } = this.props
+    const { copying, hardConflicts, softConflicts, error, selectedWorkspaceId, additionalDeletions } = this.state
+    const moreToDelete = !!additionalDeletions.length
+    const warningStyle = {
+      border: `1px solid ${colors.orange[1]}`, borderLeft: 'none', borderRight: 'none',
+      backgroundColor: colors.orange[4],
+      padding: '1rem 1.25rem', margin: '0 -1.25rem',
+      color: colors.orange[0], fontWeight: 'bold', fontSize: 12
+    }
+    const errorStyle = {
+      ...warningStyle,
+      border: `1px solid ${colors.red[1]}`,
+      backgroundColor: colors.red[4],
+      color: colors.red[0]
+    }
+
+    const errors = validate(
+      { selectedWorkspaceId },
+      { selectedWorkspaceId: { presence: true } },
+      { prettify: v => (validate.prettify(v)) }
```

Comment:
nit: You only have to specify this option if you're overriding the default prettify function somehow.